### DISCUSSION TITLE
How Can i add Channel Id , Duration , Video id in commands/play.js ?

### DISCUSSION THREAD

**Discussion by fire4912** [eritislami/evobot#225] [Q&A]
How Can i add Channel Id , Duration , Video id ? like Rythm Queue

**Comment by Artistic18** [comment]
For getting the video duration you can simply simply output `${parseInt(song.duration/60)}:${song.duration%60}`
For Video Id you can simply output `${song.url}`

**Reply by fire4912** [reply]
Is there any way to add Youtube Video's Thumbnail in Bot's Embed Thumbnail 
?

**Reply by xenodirt** [reply]
> Is there any way to add Youtube Video's Thumbnail in Bot's Embed Thumbnail
> ?

You can obtain the thumbnail through an embedded message.
`.setImage(song.thumbnail)`
or
`.setThumbnail(song.thumbnail)`

**Reply by kasepta** [reply]
@Artistic18 how when song duration until 1h? and the format to hh:mm:ss ?

**Reply by kasepta** [reply]
> .setImage(song.thumbnail)

why the image not showing? @Xenodirt

**Reply by fire4912** [reply]
> > Is there any way to add Youtube Video's Thumbnail in Bot's Embed Thumbnail
> > ?
> 
> You can obtain the thumbnail through an embedded message.
> `.setImage(song.thumbnail)`
> or
> `.setThumbnail(song.thumbnail)`

I tried this but it doesn't supports and shows nothing in result.

**Reply by brandonst** [reply]
song.thumbnail doesnt exist, the queue construct is only
```js
song = {
    title: songInfo.videoDetails.title,
    url: songInfo.videoDetails.video_url,
    duration: songInfo.videoDetails.lengthSeconds
}
```
But, you can get the image by using the url `https://img.youtube.com/vi/${video_id}/mqdefault.jpg`
just replace ${video_id} with the videos id

**Reply by kasepta** [reply]
thanks.. @brandonst do u know how to count the queue ?

**Reply by brandonst** [reply]
do you mean the song queue length? if so, you would have to first get the queue by doing, for example,
`const serverQueue = message.client.queue.get(message.guild.id)`
then you can get the length with `serverQueue.songs.length`

**Reply by kasepta** [reply]
thankyou !, i have last question. how to edit embed message queu, when i add the song the queue will automatic change.
like Hydra bot @brandonst

**Reply by xenodirt** [reply]
> > .setImage(song.thumbnail)
> 
> why the image not showing? @Xenodirt

Sorry for the late reply...
You have to define `song.thumbnail` first.
```
song = {
          title: songInfo.videoDetails.title,
          url: songInfo.videoDetails.video_url,
          duration: songInfo.videoDetails.lengthSeconds,
          thumbnail: songInfo.videoDetails.thumbnail.thumbnails[3].url
};
```

**Comment by Artistic18** [comment]
@kasepta use this ${new Date(song.duration * 1000).toISOString().substr(11,8)}

**Reply by kasepta** [reply]
when i add the playlist, the duration change to 23:59:59 @Artistic18

**Reply by Artistic18** [reply]
@kasepta Srry it took me a lot of time to figure out, it seems we cant fetch the duration just by using video.DurationSeconds in playlist.js.
There we have to add each song of the playlist individually.
In playlist.js add these lines and make the necessary changes  :

const vidpromise = [];
        for (const v of videos)
          v.title !== 'Private video' && v.thumbnails && vidpromise.push(youtube.getVideoByID(v.id));
       
 const youtubeVideos = await Promise.all(vidpromise);

const newList = youtubeVideos.map((video) => {
            return (song = {
                title: video.title,
                url: video.url,
                duration: video.durationSeconds,
                thumbnail: video.thumbnails.medium.url
            });
        });

**Comment by kasepta** [comment]
Anyone know? i try to edit embed message when volume up / down. but idk what i must edit, in here or where?
```js
        case "🔊":
          reaction.users.remove(user).catch(console.error);
          if (!canModifyQueue(member)) return;
          if (queue.volume + 10 >= 100) queue.volume = 100;
          else queue.volume = queue.volume + 10;
          queue.connection.dispatcher.setVolumeLogarithmic(queue.volume / 100);
          queue.textChannel
            .send(`${user} 🔊 increased the volume, the volume is now ${queue.volume}%`)
            .then(m => m.delete({ timeout: 3000 }))
            .catch(console.error);
          break;
```

**Reply by fire4912** [reply]
> Anyone know? i try to edit embed message when volume up / down. but idk what i must edit, in here or where?
> 
> ```js
>         case "🔊":
>           reaction.users.remove(user).catch(console.error);
>           if (!canModifyQueue(member)) return;
>           if (queue.volume + 10 >= 100) queue.volume = 100;
>           else queue.volume = queue.volume + 10;
>           queue.connection.dispatcher.setVolumeLogarithmic(queue.volume / 100);
>           queue.textChannel
>             .send(`${user} 🔊 increased the volume, the volume is now ${queue.volume}%`)
>             .then(m => m.delete({ timeout: 3000 }))
>             .catch(console.error);
>           break;
> ```

Do you want to set the Volume reply as a Embed Msg?

**Reply by kasepta** [reply]
no, the bot edit the old message embed, to new, like Hydra music

**Comment by velo-prash98** [comment]
Can anyone help how to add seek command in the bot and also it doesnt read the spotify urls, can anyone help? @fire4912 @Bikey29 @Artistic18

**Reply by fire4912** [reply]
> Can anyone help how to add seek command in the bot and also it doesnt read the spotify urls, can anyone help? @fire4912 @Bikey29 @Artistic18

You do it by adding SPotify api . and seek command maybe added please check previous commits .

**Reply by Artistic18** [reply]
@velo-prash98 I already posted an answer for seek command in your discussion.
